net_connection/src/net_connection_thread.rs

```diff
@@ -32,16 +32,21 @@ impl NetConnectionThread {
     /// stop (join) the worker thread
     pub fn stop(self) -> NetResult<()> {
         self.keep_running.store(false, Ordering::Relaxed);
-        match self.thread.join() {
-            Ok(_) => Ok(()),
-            Err(_) => {
-                bail!("NetConnectionThread failed to join on stop() call");
-            }
+        if let Err(_) = self.thread.join() {
```

Comment:
Again, please use the `is_err()` method

Reply:
I feel like this might be subjective style, no? If you use a `_` doesn't the compiler optimize out the unused stack variable?

Went ahead and changed this, incase the optimizer isn't that smart.